gammapy/detect/cwt.py

```diff
@@ -188,127 +197,159 @@ def set_data(self, counts, background):
         self.model = np.zeros(shape_2d)
         self.approx = np.zeros(shape_2d)
 
-        shape_3d = self.nscale, shape_2d[0], shape_2d[1]
+        shape_3d = self.n_scale, shape_2d[0], shape_2d[1]
         self.transform = np.zeros(shape_3d)
         self.error = np.zeros(shape_3d)
         self.support = np.zeros(shape_3d)
 
-    def do_transform(self):
+    def _execute_iteration(self):
+        """Do one iteration of the algorithm."""
+        self._transform()
+        self._compute_support()
+        result = self._inverse_transform()
+        return result
+
+    def _transform(self):
         """
         Do the transform itself.
 
-        TODO: document. rename?
+        The transform is made using 'scipy.signal.fftconvolve'.
+
+        TODO: document.
         """
         from scipy.signal import fftconvolve
+
         total_background = self.model + self.background + self.approx
         excess = self.counts - total_background
 
+        log.info('Computing transform and error')
         for idx_scale, kern in self.kernbase.items():
-            log.info('Computing transform and error')
             self.transform[idx_scale] = fftconvolve(excess, kern, mode='same')
             self.error[idx_scale] = np.sqrt(fftconvolve(total_background, kern ** 2, mode='same'))
 
+        log.info('Computing approx and approx_bkg')
         self.approx = fftconvolve(self.counts - self.model - self.background,
                                   self.kern_approx, mode='same')
         self.approx_bkg = fftconvolve(self.background, self.kern_approx, mode='same')
 
-    def compute_support(self, nsigma=2.0, nsigmap=4.0, remove_isolated=True):
+    def _compute_support(self):
         """
         Compute the multiresolution support with hard sigma clipping.
 
         Imposing a minimum significance on a connected region of significant pixels
-        (i.e. source detection)
-
-        Parameters
-        ----------
-        TODO: document
-
-        Returns
-        -------
-        TODO: for now it's stored as `self.support`. Maybe return
+        (i.e. source detection).
+
+        TODO: document?
```

Comment:
I improve this method and think that it will be easier to understand what's going on here after my modifications.  @cdeil @adonath could you review this to check the logic? And than I write the docs to this method.